### PR TITLE
Add Github Actions support

### PR DESCRIPTION
**Pull request by ibrahim0814**
# Purpose

Currently there's no way for the node uploader to detect Actions CI/CD pipelines. This PR adds support for it.

**Note**: There's currently no way to obtain a build_url or build number from the Actions environment variables. That's why they've been excluded in this PR.

### CODE REVIEW

**Comment by ibrahim0814** [comment]
@eddiemoore Do you know why these tests are failing? I can't really tell why from the output on Travis

File: test/services/github_actions.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    process.env.GITHUB_ACTIONS = '1'
+    expect(github_actions.detect()).to.be(true)
+  })
+
+  it('can get GitHub Actions env info', function() {
+    process.env.GITHUB_SHA = '743b04806ea677403aa2ff26c6bdeb85005de658'
+    process.env.GITHUB_REPOSITORY = 'codecov/codecov-repo'
+    process.env.GITHUB_REF = 'refs/heads/master'
+
+    expect(github_actions.configuration()).to.eql({
```
Comment (eddiemoore) [inline test/services/github_actions.test.js]: `.toEqual`

**Comment by eddiemoore** [comment]
Sorry it's taken me so long to review this because of my current work.

**Comment by ibrahim0814** [comment]
@eddiemoore No worries! Fixed the jest assertions and Travis is now passing.

File: lib/services/github_actions.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module.exports = {
+    detect: function() {
+      return !!process.env.GITHUB_ACTIONS
+    },
+  
+    configuration: function() {
+      console.log('    GitHub Actions CI Detected')
+      return {
+        service: 'github_actions',
```
Comment (eddiemoore) [inline lib/services/github_actions.js]: Found out it needs to be `github-actions`

File: test/services/github_actions.test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    expect(github_actions.detect()).toBe(true)
+  })
+
+  it('can get GitHub Actions env info', function() {
+    process.env.GITHUB_SHA = '743b04806ea677403aa2ff26c6bdeb85005de658'
+    process.env.GITHUB_REPOSITORY = 'codecov/codecov-repo'
+    process.env.GITHUB_REF = 'refs/heads/master'
+
+    expect(github_actions.configuration()).toEqual({
+      service: 'github_actions',
```
Comment (eddiemoore) [inline test/services/github_actions.test.js]: Same thing here

**Comment by ibrahim0814** [comment]
Corrected the service name!

**Comment by JustinBeckwith** [comment]
Greetings folks!  This is a change that would make our lives much easier :) Anything I can do to help it get merged?

**Comment by ibrahim0814** [comment]
Accidentally closed this PR, sorry about that. @eddiemoore, is this ready to merge?

**Review by thomasrockhu — APPROVED** [review]

**Comment by thomasrockhu** [comment]
As always, thanks @ibrahim0814

**Pull request merged**